spectec/spec/7-module.watsup

```diff
@@ -0,0 +1,134 @@
+;; Allocation
+
+
+def $funcs(externval*) : funcaddr*
+def $globals(externval*) : globaladdr*
+def $tables(externval*) : tableaddr*
+def $mems(externval*) : memaddr*
+
+def $funcs(externval*) = [fa|FUNC fa <- externval]
+def $globals(externval*) = [ga|GLOBAL ga <- externval]
+def $tables(externval*) = [ta|TABLE ta <- externval]
+def $mems(externval*) = [ma|MEM ma <- externval]
```

Comment:
I suggest new syntax for order preserving filter in the spec.
<img width="720" alt="image" src="https://github.com/Wasm-DSL/spectec/assets/50018375/76ebb1b7-9728-44c7-af0d-50929671bf28">

Or, we can use recursion
```
def $funcs(externval*) = funcaddr*
def $funcs(epsilon) = epsilon
def $funcs(FUNC fa externval'*) = fa $funcs(externval'*)
def $funcs(externval externval'*) = $funcs(externval'*)
  -- otherwise
```

Reply:
I updated the syntax in [236de04](https://github.com/Wasm-DSL/spectec/pull/32/commits/236de048db80bbad5ee3d581f273c5eacbb58a64).
```
def $funcs(externval*) = `[fa|FUNC fa <- externval]
def $globals(externval*) = `[ga|GLOBAL ga <- externval]
def $tables(externval*) = `[ta|TABLE ta <- externval]
def $mems(externval*) = `[ma|MEM ma <- externval]
```

Reply:
Makes sense, though extending the DSL may be quite a cost for a one-off auxiliary definition like this. Let's perhaps discuss this on Thursday.